### ISSUE TITLE
Aggregated Feedback when bundling Vue with Rolldown

### ISSUE THREAD

**Issue by yyx990803** [rolldown/rolldown#1264]
I successfully bundled Vue (currently only the main `vue` package) with Rolldown (0.10.2 published on npm): https://github.com/vuejs/core/compare/rolldown

### Highlights

- I was able to remove `commonjs`, `node-resolve` and `alias` plugins due to built-in support, and the rest of the plugins just worked!
- Bundled file mostly works as expected, very similar to existing Rollup output

### Small Issues / Improvements to be made

- [x] #1274
- [x] #1266
- [x] #1267
- [x] #1268
- [x] #1269
- [x] #1270
- [x] If `output.banner` is specified, it throws the following error:
   
  ```
  [10:49:23 AM]  ERROR  Rolldown internal error: InvalidArg, Unknown return value. Cannot convert to 
  core::option::Option<alloc::string::String>.
  ```

  The error isn't telling where the problem is so I had to comment off the options one by one to find out.

- [x] #1271
- [x] #1272
- [x] #1273

**Comment by hyf0** [comment]
>  If output.banner is specified, it throws the following error:

This is fixed in 0.10.3.

**Comment by hyf0** [comment]
> Running rolldown from the CLI doesn't exit after the build ends.

has a workaround fix which is already landed in 0.10.3. The final fix is tracking in #1274 now.

**Comment by hyf0** [comment]
> Bundled files contains unnecessary 'use strict' directive for each bundled module

This is also fixed in 0.10.3. 

~This seems to be an old behavior of oxc-codegen, which will emits `"use strict"` by default. The newest oxc-codegen doesn't emit `"use strict"` anymore. Judging by the result, the issue is considered to be fixed.  @Boshen cc~

However, the issue become if rolldown emits `use strict` when it should. I will create an another issue to track this problem.

---edited

Rolldown will render \`"use strict"\` if the original code contains \`"use strict"\` in 0.10.3. This is not expected and will be fixed in #1277.

**Comment by Boshen** [comment]
Vue ecosystem CI is green, and all remaining issues have tracking issues, closing.
